### PR TITLE
Add copr config template

### PR DESCRIPTION
**Pull request by ehelms**
@GregSutcliffe Check me if I am wrong in how we handle configuring files with secrets here. I think these then get defined in our Foreman instance?

### CODE REVIEW

File: puppet/modules/slave/manifests/init.pp [hunk 1]
```diff
@@ -393,6 +393,13 @@
     group  => 'jenkins',
     source => 'puppet:///modules/slave/katello-ca.cert',
   }
+  file { '/home/jenkins/.config/copr':
+    ensure   => file,
+    mode     => '0644',
```
Comment (ekohl) [inline puppet/modules/slave/manifests/init.pp]: I think 640 might be better since it has tokens

File: puppet/modules/slave/templates/copr.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[copr-cli]
+login = <%= @copr_login %>
```
Comment (ekohl) [inline puppet/modules/slave/templates/copr.erb]: Where do these variables come from? Global variables?

File: puppet/modules/slave/templates/copr.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+username = <%= copr_username %>
```
Comment (ekohl) [inline puppet/modules/slave/templates/copr.erb]: Do these also need ```@```?

File: puppet/modules/slave/templates/copr.erb (hunk 2 above)
Comment (ehelms) [reply puppet/modules/slave/templates/copr.erb]: I *think* so, but hoping @GregSutcliffe can confirm

Comment (GregSutcliffe) [reply puppet/modules/slave/templates/copr.erb]: Foreman can hold secure variables and pas them in as smart class parameters, yes. Just set sane defaults here.

File: puppet/modules/slave/manifests/init.pp (hunk 1 above)
Comment (GregSutcliffe) [reply puppet/modules/slave/manifests/init.pp]: +1

**Comment by GregSutcliffe** [comment]
Does this have to live on the slaves? I trust all our sponsors but technically that's a large group of people who could get our certificate. Is there another way to pass this in?

**Comment by ehelms** [comment]
The only other option I know about is to create a temporary config file in the CI job, and point copr-cli at that, then clean it up with the workspace.

**Comment by GregSutcliffe** [comment]
I guess that's not much better, given it'd be predictable. Does Copr handling signing or is that step done elsewhere?

**Comment by ehelms** [comment]
I'll give the above a try before we go this route.

**Comment by ehelms** [comment]
RPM signing? Copr handles that. The temporary config file would only be persistent for a short period, however, this is proving difficult to do with our use of tito on top of things. Unfortunately, Koji tends to work this same way, so we have this issue regardless.

**Comment by GregSutcliffe** [comment]
Fair enough. I was under the impression that the releases were signed separately.

**Comment by ehelms** [comment]
@ekohl Updated

**Comment by ekohl** [comment]
I wonder if we could solve the credentials issue is with splitting the builds: jenkins can only builds to the scratch repo(s) and handle production releases through github webhooks to copr.

**Pull request merged**